### PR TITLE
NIFI-9761 Correct PeerChannel processing for TLS 1.3

### PR DESCRIPTION
**Pull request by exceptionfactory**
#### Description of PR

NIFI-9761 Corrects `PeerChannel` handling of TLS 1.3 post-handshake messages, avoiding unexpected behavior when attempting to read from the underlying `SocketChannel` after previous calls have already put data in the stream buffer. This resolves a latent issue with Load-Balanced Connections and TLS 1.3 communication.

The new `TestPeerChannel` leverages a simple socket server based on Netty to confirm expected behavior for both TLS 1.2 and TLS 1.3.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [X] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [X] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by joewitt** [comment]
such a great find/fix.

**Comment by exceptionfactory** [comment]
Further testing with the `LoadBalanceIT` in `nifi-system-test-suite` uncovered an issue with an unnecessary call to `streamBuffer.compact()`, as well as missing `SSLEngine` close notification processing.  The current version includes additional trace logging.

**Review by greyp9 — APPROVED** [review]
Given our understanding of the nature of this issue, these changes look good.  I've done some testing of `LoadBalanceIT`, and the behavior seems to align much better with what is expected of a TLSv1.3 connection peer.  

Nice unit test!  That will come in handy if we need to revisit this implementation in the future.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/queue/clustered/client/async/nio/PeerChannel.java [hunk 1]
```diff
@@ -79,9 +114,9 @@ public OptionalInt read() throws IOException {
         singleByteBuffer.clear();
         final int bytesRead = read(singleByteBuffer);
         if (bytesRead < 0) {
-             return OptionalInt.of(-1);
+             return OptionalInt.of(END_OF_FILE);
         }
-        if (bytesRead == 0) {
+        if (bytesRead == EMPTY_BUFFER) {
```
[the remark this answers is not in the document]
Comment (exceptionfactory) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/queue/clustered/client/async/nio/PeerChannel.java:119]: This change was just for variable reuse, but that seems like a reasonable improvement.

**Review by exceptionfactory — COMMENTED** [review]
Thanks for the feedback and testing @greyp9! I will add a comment and make the adjustment noted.

**Comment by markap14** [comment]
That's a great catch @exceptionfactory , thanks for finding & fixing that! I'm a +1 will merge

**Pull request merged**